zerver/actions/message_edit.py

```diff
@@ -1351,6 +1351,8 @@ def check_update_message(
     It returns the number changed.
     """
     message = access_message(user_profile, message_id, lock_message=True)
+    if message.recipient.type == Recipient.STREAM:
+        access_stream_by_id(user_profile, message.recipient.type_id)
```

Comment:
Can you explain the strategy here? I don't understand what you're doing.

Reply:
I see, the point here is that 

Let's make this work as an `allow_archived_channel` parameter to `access_message` that defaults to `true`, and pass that parameter in the code paths that want it.